src/services/actions/CliActions.ts

```diff
@@ -479,21 +480,21 @@ export class CliActions {
       cancellable: true
     }, async (progress: Progress<{ message?: string; increment?: number }>) => {
       try {
-        const result = await CliExecuter.execute('spfx project upgrade', 'md');
-
-        if (result.stdout) {
-          // Create a file to allow the Markdown preview to correctly open the linked/referenced files
-          let savePath = wsFolder?.uri.fsPath;
-
-          if (savePath && TeamsToolkitIntegration.isTeamsToolkitProject) {
-            savePath = join(savePath, 'src');
-          }
+        const projectUpgradeOutputMode: string = getExtensionSettings('projectUpgradeOutputMode', 'both');
+        let resultMd: any;
+
+        if (projectUpgradeOutputMode === 'markdown') {
+          resultMd = await CliExecuter.execute('spfx project upgrade', 'md');
+          CliActions.handleMarkdownResult(resultMd, wsFolder, 'upgrade');
+        } else if (projectUpgradeOutputMode === 'code tour') {
+          await CliExecuter.execute('spfx project upgrade', 'tour');
+          CliActions.handleTourResult(wsFolder, 'upgrade');
+        } else {
+          resultMd = await CliExecuter.execute('spfx project upgrade', 'md');
+          await CliExecuter.execute('spfx project upgrade', 'tour');
```

Comment:
I was wondering if we could just have `if` blocks in which we check for either the specific type/method, like `markdown` or `tour`  and check for `both`. In this case we don't need to check explicitly for `both` and we will avoid code repetition. What do you think? I guess it should work

```suggestion
        if (projectUpgradeOutputMode === 'markdown' || projectUpgradeOutputMode === 'both') {
          resultMd = await CliExecuter.execute('spfx project upgrade', 'md');
          CliActions.handleMarkdownResult(resultMd, wsFolder, 'upgrade');
        } 
        
        if (projectUpgradeOutputMode === 'code tour' || projectUpgradeOutputMode === 'both') {
          await CliExecuter.execute('spfx project upgrade', 'tour');
          CliActions.handleTourResult(wsFolder, 'upgrade');
        }```